src/contracts/core/DelegationManager.sol

```diff
@@ -510,199 +505,178 @@ contract DelegationManager is
         delegatedTo[staker] = operator;
         emit StakerDelegated(staker, operator);
 
+        // read staker's delegatable shares and strategies to add to operator's delegatedShares
+        // and also update the staker depositScalingFactor for each strategy
         (IStrategy[] memory strategies, uint256[] memory shares) = getDelegatableShares(staker);
+        uint64[] memory totalMagnitudes = allocationManager.getTotalMagnitudes(operator, strategies);
 
-        for (uint256 i = 0; i < strategies.length;) {
+        for (uint256 i = 0; i < strategies.length; ++i) {
             // forgefmt: disable-next-item
-            _increaseOperatorShares({
+            _increaseDelegation({
                 operator: operator, 
                 staker: staker, 
-                strategy: strategies[i], 
-                shares: shares[i]
+                strategy: strategies[i],
+                existingPrincipalShares: 0,
+                addedShares: shares[i],
+                totalMagnitude: totalMagnitudes[i]
             });
-
-            unchecked {
-                ++i;
-            }
         }
     }
 
     /**
-     * @dev commented-out param (middlewareTimesIndex) is the index in the operator that the staker who triggered the withdrawal was delegated to's middleware times array
-     * This param is intended to be passed on to the Slasher contract, but is unused in the M2 release of these contracts, and is thus commented-out.
+     * @dev This function completes a queued withdrawal for a staker.
+     * This will apply any slashing that has occurred since the the withdrawal was queued. By multiplying the withdrawl's
+     * delegatedShares by the operator's total magnitude for each strategy
+     * If receiveAsTokens is true, then these shares will be withdrawn as tokens.
+     * If receiveAsTokens is false, then they will be redeposited according to the current operator the staker is delegated to,
+     * and added back to the operator's delegatedShares.
      */
     function _completeQueuedWithdrawal(
         Withdrawal calldata withdrawal,
         IERC20[] calldata tokens,
-        uint256, /*middlewareTimesIndex*/
         bool receiveAsTokens
     ) internal {
+        require(tokens.length == withdrawal.strategies.length, InputArrayLengthMismatch());
+        require(msg.sender == withdrawal.withdrawer, WithdrawerNotCaller());
         bytes32 withdrawalRoot = calculateWithdrawalRoot(withdrawal);
+        require(pendingWithdrawals[withdrawalRoot], WithdrawalNotQueued());
+
+        // TODO: is there a cleaner way to do this?
+        uint32 completableTimestamp = _checkCompletability(withdrawal.startTimestamp);
+        // read delegated operator's totalMagnitudes at time of withdrawal to convert the delegatedShares to shared
```

Comment:
typo: `shared` => `shares`
also technical nit but this is not `at the time of withdrawal` it is `at the earliest time that the withdrawal could be completed`